src/java.desktop/macosx/native/libawt_lwawt/awt/AWTWindow.m

```diff
@@ -1024,7 +1024,11 @@ - (void)sendEvent:(NSEvent *)event {
             NSRect contentRect = [self.nsWindow contentRectForFrameRect:frame];
 
             // Check if the click happened in the non-client area (title bar)
-            if (p.y >= (frame.origin.y + contentRect.size.height)) {
+            // Also, non-client area includes the edges at left, right and botton of frame
+            if ((p.y >= (frame.origin.y + contentRect.size.height)) ||
+                (p.x >= (frame.origin.x + contentRect.size.width - 3)) ||
+                (fabs(frame.origin.x - p.x) < 3) ||
+                (fabs(frame.origin.y - p.y) < 3)) {
```

Comment:
Does `fabs` use float? It makes code more compact but it may be imprecise and less performant than integer arithmetics.

Reply:
Yes..https://man7.org/linux/man-pages/man3/fabs.3.html and [CGFLoat ](https://developer.apple.com/documentation/corefoundation/cgfloat?language=objc) can be double/float and fabs can accomodate both, in this case it uses double...We have tested on 3 different systems without any issue so would keep it for now until we find any adverse effect..

Reply:
I see, the coordinates are floats, then it's perfectly fine. I assumed the coordinates were integers.